src/riak_ensemble_manager.erl

```diff
@@ -369,6 +375,13 @@ handle_call({peer_info, Ensemble, Peer}, From, State) ->
             {noreply, State}
     end;
 
+handle_call({subscribe, Callback}, _, State=#state{subscribers=Subscribers}) ->
+    %% TODO: Change subscriber API to prevent multiple subscriptions with the same callback.
+    Subscribers2 = [Callback|Subscribers],
+    application:set_env(riak_ensemble, subscribers, Subscribers2),
```

Comment:
In this case, this is the least frictionless way to keep some global state to persist even in the even of a crash of ensemble. Safety wise, it appears to be perfectly fine.